### PR TITLE
MESH-1656/bridge freezing permissions

### PR DESCRIPTION
**Pull request by Neopallium**
### new features

- Add support for multiple freeze admin accounts to the bridge pallet.

### modified API

- New extrinsics `add_freeze_admin` and `remove_freeze_admin` to `bridge` pallet.  These can only be called by the current bridge admin to add/remove freeze admins.
- New events `FreezeAdminAdded` and `FreezeAdminRemoved`.

### modified logic

- Bridge extrinsic `freeze` can now be called by any of the freeze admin accounts.

### CODE REVIEW

**Comment by Neopallium** [comment]
Also some cleanup of compile warnings from the tests.

File: pallets/bridge/src/lib.rs [hunk 1]
```diff
@@ -843,8 +879,22 @@ impl<T: Config> Module<T> {
...
+
+    fn base_remove_freeze_admin(origin: T::Origin, freeze_admin: T::AccountId) -> DispatchResult {
+        let did = Self::ensure_admin_did(origin)?;
+        <FreezeAdmins<T>>::remove(freeze_admin.clone());
+        Self::deposit_event(RawEvent::FreezeAdminRemoved(did, freeze_admin));
+        Ok(())
+    }
+
+    fn set_freeze(origin: T::Origin, freeze: bool) -> DispatchResult {
+        let did = Self::ensure_freeze_admin_did(origin)?;
```
Comment (Centril) [inline pallets/bridge/src/lib.rs:897]:
```suggestion
        let did = if freeze {
            Self::ensure_freeze_admin_did(origin)
        } else {
            Self::ensure_admin_did(origin)
        }?;
```
Spec says:
> NB - these “freeze admin” addresses should only be able to freeze the entire bridge, not freeze individual transactions and **not be able to unfreeze the bridge.**

Comment (Neopallium) [reply pallets/bridge/src/lib.rs:897]: Ah.  I forgot that condition.

File: pallets/runtime/tests/src/bridge.rs [hunk 2]
```diff
@@ -273,37 +273,26 @@ fn do_test_freeze_admins(_signers: &[AccountId]) {
...
-            Bridge::freeze(user.origin()),
-            Error::BadAdmin
-        );
-        assert!(!Bridge::frozen());
-
-        // Use admin to freeze the bridge.
-        assert_ok!(Bridge::freeze(admin.clone()));
-        assert!(Bridge::frozen());
+        if can_freeze {
+            // User is not allowed to freeze the bridge.
```
Comment (Centril) [inline pallets/runtime/tests/src/bridge.rs:281]:
```suggestion
            // User is allowed to freeze the bridge.
```

**Review by CJP10 — APPROVED** [review]

**Review by Centril — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Pull request merged**